### DISCUSSION TITLE
(embed mode) Merge doesnt always work

### DISCUSSION THREAD

**Discussion by reinoutkoninkx** [jgraph/drawio#4468] [Q&A]
using the postMessage API, using the merge action does not always work.

It works flawlessly until the point you remove something from a diagram and then using the postMessage API to merge it back on again.

It seems as though the id's of the shapes that were removed from the diagram somewhere in the code do not get removed from some kind of list, therefore it seems to think the shape with that id is still on the diagram, causing it not to add it again after merging.

**Comment by davidjgraph** [comment]
Duplicate #4229 . Warning for breaking code of conduct by re-posting.

**Comment by alderg** [comment]
Please include the merge message and the XML of the diagram before and after removing.
